src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientProvider.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Azure.Core;
+using Azure.Data.AppConfiguration;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Constants;
+using Microsoft.Extensions.Configuration.AzureAppConfiguration.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    /// <summary>
+    /// A configuration client provider which maintains state of configuration clients and provides set of clients to use when requested.
+    /// </summary>
+    /// <remarks>
+    /// This class is not thread-safe. Since config provider does not allow multiple network requests at the same time,
+    /// there won't be multiple threads calling this client at the same time.
+    /// </remarks>
+    internal class ConfigurationClientProvider : IConfigurationClientProvider
```

Comment:
This class manages a list of configuration clients and decides which one to use. Do you feel `ConfigurationClientManager` reflects better what it does? I was confused at the beginning with the "provider" naming and thought it would provides one instance of configuration client to someone else who manages all clients.

Reply:
I think that makes more sense!